spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkPartitioningAwareScan.java

```diff
@@ -232,6 +234,13 @@ protected synchronized List<ScanTaskGroup<T>> taskGroups() {
     return taskGroups;
   }
 
+  private long targetSplitSize() {
+    long scanSize = tasks().stream().mapToLong(ScanTask::sizeBytes).sum();
```

Comment:
This would even handle runtime filtering, so that the number of splits after runtime filtering may be different.